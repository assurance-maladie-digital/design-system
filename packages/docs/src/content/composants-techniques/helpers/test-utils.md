---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
