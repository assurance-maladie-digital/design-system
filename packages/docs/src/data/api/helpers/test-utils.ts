import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
