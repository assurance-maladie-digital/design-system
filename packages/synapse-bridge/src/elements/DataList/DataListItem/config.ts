export const config = {
	icon: {
		size: 24,
		class: 'mr-4 mt-2',
	},
	chip: {
		class: 'mt-1',
	},
	actionBtn: {
		variant: 'text',
		size: 'small',
		color: 'accent',
		class: 'text-body-1 pa-0',
	}
}
