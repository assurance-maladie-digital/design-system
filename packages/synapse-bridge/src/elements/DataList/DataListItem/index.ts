export { default } from './DataListItem.vue';
