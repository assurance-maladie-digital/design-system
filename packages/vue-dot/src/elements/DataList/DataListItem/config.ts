export const config = {
	icon: {
		size: 24,
		class: 'mr-4 mt-2'
	},
	chip: {
		class: 'mt-1'
	},
	actionBtn: {
		text: true,
		small: true,
		color: 'accent',
		class: 'text-body-1 pa-0'
	}
};
