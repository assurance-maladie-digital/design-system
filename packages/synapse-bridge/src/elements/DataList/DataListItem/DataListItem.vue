<script lang="ts">
	import { defineComponent } from 'vue'

	import { config } from './config'
	import { locales } from './locales'

	import { customizable } from '@/mixins/customizable'

	import { useTheme } from 'vuetify'

	export default defineComponent({
		mixins: [customizable(config)],
		props: {
			label: {
				type: String,
				required: true,
			},
			value: {
				type: [String, Number],
				default: undefined,
			},
			action: {
				type: String,
				default: undefined,
			},
			placeholder: {
				type: String,
				default: locales.placeholder,
			},
			chip: {
				type: Boolean,
				default: false,
			},
			icon: {
				type: String,
				default: undefined,
			},
			row: {
				type: Boolean,
				default: false,
			},
			renderHtmlValue: {
				type: Boolean,
				default: false,
			},
		},
		computed: {
			labelColor(): string {
				const theme = useTheme()

				return theme.current.value.dark ? 'rgba(255, 255, 255, .7)' : 'rgba(0, 0, 0, .6)'
			},
			itemValue(): string {
				if (typeof this.value === 'number') {
					return isNaN(this.value) ? this.placeholder : this.value.toString()
				}

				return this.value || this.placeholder
			}
		}
	})
</script>

<template>
	<li class="vd-data-list-item d-flex flex-wrap">
		<slot name="icon">
			<VIcon
				v-if="icon"
				v-bind="options.icon"
			>
				{{ icon }}
			</VIcon>
		</slot>

		<div class="vd-data-list-item-content">
			<div :class="{ 'vd-row': row }">
				<div
					class="vd-data-list-item-label text-caption"
					:style="{ color: labelColor }"
				>
					{{ label }}
				</div>

				<div class="vd-data-list-item-value">
					<slot
						name="value"
						v-bind="{ itemValue }"
					>
						<VChip
							v-if="chip"
							v-bind="options.chip"
						>
							{{ itemValue }}
						</VChip>

						<span
							v-else-if="renderHtmlValue"
							class="text-body-1"
							v-html="itemValue"
						/>

						<span v-else class="text-body-1" v-text="itemValue" />
					</slot>
				</div>
			</div>

			<slot name="action">
				<VBtn
					v-if="action"
					v-bind="options.actionBtn"
					class="vd-data-list-item-action-btn"
					@click="$emit('click:action')"
				>
					{{ action }}
				</VBtn>
			</slot>
		</div>
	</li>
</template>

<style lang="scss" scoped>
	.vd-row {
		display: flex;
		flex-wrap: wrap;

		.vd-data-list-item-label {
			align-self: center;

			&::after {
				content: ":";
				margin: 0 4px;
			}
		}
	}

	.vd-data-list-item-action-btn.v-btn {
		min-width: 0;
		margin: 0 -1px;
	}

	.v-icon.v-theme--light {
		color: rgba(0, 0, 0, 0.54);
	}
</style>
