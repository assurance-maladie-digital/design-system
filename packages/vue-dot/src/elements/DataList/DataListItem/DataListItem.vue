<template>
	<li class="vd-data-list-item d-flex flex-wrap">
		<slot name="icon">
			<VIcon
				v-if="icon"
				v-bind="options.icon"
			>
				{{ icon }}
			</VIcon>
		</slot>

		<div class="vd-data-list-item-content">
			<div :class="{ 'vd-row': row }">
				<div
					class="vd-data-list-item-label text-caption"
					:style="{ color: labelColor }"
				>
					{{ label }}
				</div>

				<div class="vd-data-list-item-value">
					<slot
						name="value"
						v-bind="{ itemValue }"
					>
						<VChip
							v-if="chip"
							v-bind="options.chip"
						>
							{{ itemValue }}
						</VChip>

						<span
							v-else-if="renderHtmlValue"
							class="text-body-1"
							v-html="itemValue"
						/>

						<span
							v-else
							class="text-body-1"
							v-text="itemValue"
						/>
					</slot>
				</div>
			</div>

			<slot name="action">
				<VBtn
					v-if="action"
					v-bind="options.actionBtn"
					class="vd-data-list-item-action-btn"
					@click="$emit('click:action')"
				>
					{{ action }}
				</VBtn>
			</slot>
		</div>
	</li>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { config } from './config';
	import { locales } from './locales';

	import { customizable } from '../../../mixins/customizable';

	const Props = Vue.extend({
		props: {
			label: {
				type: String,
				required: true
			},
			value: {
				type: [String, Number],
				default: undefined
			},
			action: {
				type: String,
				default: undefined
			},
			placeholder: {
				type: String,
				default: locales.placeholder
			},
			chip: {
				type: Boolean,
				default: false
			},
			icon: {
				type: String,
				default: undefined
			},
			row: {
				type: Boolean,
				default: false
			},
			renderHtmlValue: {
				type: Boolean,
				default: false
			}
		}
	});

	const MixinsDeclaration = mixins(Props, customizable(config));

	@Component
	export default class DataListItem extends MixinsDeclaration {
		get labelColor(): string {
			return this.$vuetify.theme.dark ? 'rgba(255, 255, 255, .7)' : 'rgba(0, 0, 0, .6)';
		}

		get itemValue(): string {
			if (typeof this.value === 'number') {
				return isNaN(this.value) ? this.placeholder : this.value.toString();
			}

			return this.value || this.placeholder;
		}
	}
</script>

<style lang="scss" scoped>
	.vd-row {
		display: flex;
		flex-wrap: wrap;

		.vd-data-list-item-label {
			align-self: center;

			&::after {
				content: ":";
				margin-right: 4px;
			}
		}
	}

	.vd-data-list-item-action-btn.v-btn {
		min-width: 0;
		margin: 0 -1px;
	}
</style>
