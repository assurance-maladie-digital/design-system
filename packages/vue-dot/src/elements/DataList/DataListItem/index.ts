import DataListItem from './DataListItem.vue';

export default DataListItem;
