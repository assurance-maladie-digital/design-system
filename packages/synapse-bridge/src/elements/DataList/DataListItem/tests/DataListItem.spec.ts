import { describe, it, expect } from "vitest";
import { shallowMount } from "@vue/test-utils";
import { vuetify } from "../../../../../tests/unit/setup";

import DataListItem from "../";

describe("DataListItem", () => {
	it("renders correctly", () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Test",
			},
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper).toMatchSnapshot();
	});

	it("renders correctly with a value", () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Test",
				value: "value",
			},
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper).toMatchSnapshot();
	});

	it("renders correctly a value with HTML as text", () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Name",
				value: "Paul<br> Dupont",
			},
			global: {
				plugins: [vuetify],
			},
		});

		const elValue = wrapper.find(".vd-data-list-item-value span");
		expect(elValue.text()).toBe("Paul<br> Dupont");
	});

	it("renders correctly a value as plain HTML", () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Name",
				value: "Paul<br> Dupont",
				renderHtmlValue: true,
			},
			global: {
				plugins: [vuetify],
			},
		});

		const elValue = wrapper.find(".vd-data-list-item-value span");
		expect(elValue.text()).toBe("Paul Dupont");
	});

	it("renders correctly value in a chip", () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Test",
				value: "value",
				chip: true,
			},
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper).toMatchSnapshot();
	});

	it("renders correctly with an action", () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Test",
				action: "action",
			},
			global: {
				plugins: [vuetify],
			},
		});

		expect(wrapper).toMatchSnapshot();
	});

	it("emits click:action event when the action button is pressed", async () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Test",
				value: "value",
				action: "Action",
			},
			global: {
				plugins: [vuetify],
			},
		});

		const actionBtn = wrapper.find(".vd-data-list-item-action-btn");
		expect(actionBtn.exists()).toBe(true);

		actionBtn.trigger("click");

		await wrapper.vm.$nextTick();

		expect(wrapper.emitted("click:action")).toBeTruthy();
	});

	it("renders correctly in row mode", () => {
		const wrapper = shallowMount(DataListItem, {
			propsData: {
				label: "Test",
				value: "value",
				action: "Action",
				row: true,
			},
			global: {
				plugins: [vuetify],
			},
		});

		const elExists = wrapper.find(".vd-row").exists();
		expect(elExists).toBe(true);

		expect(wrapper).toMatchSnapshot();
	});
});
