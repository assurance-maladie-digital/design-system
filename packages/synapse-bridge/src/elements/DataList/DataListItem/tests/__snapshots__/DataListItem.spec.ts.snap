// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataListItem > emits click:action event when the action button is pressed 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">value</span></div>
    </div>
    <v-btn-stub data-v-55bfbadb="" color="accent" class="text-body-1 pa-0 vd-data-list-item-action-btn" symbol="Symbol(vuetify:v-btn-toggle)" flat="false" icon="false" block="false" slim="false" stacked="false" ripple="true" border="false" density="default" disabled="false" loading="false" tile="false" replace="false" exact="false" size="small" tag="button" variant="text"></v-btn-stub>
  </div>
</li>
`;

exports[`DataListItem > renders correctly 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">…</span></div>
    </div>
    <!--v-if-->
  </div>
</li>
`;

exports[`DataListItem > renders correctly a value as plain HTML 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Name</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">Paul<br> Dupont</span></div>
    </div>
    <!--v-if-->
  </div>
</li>
`;

exports[`DataListItem > renders correctly a value with HTML as text 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Name</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">Paul&lt;br&gt; Dupont</span></div>
    </div>
    <!--v-if-->
  </div>
</li>
`;

exports[`DataListItem > renders correctly in dark mode 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(255, 255, 255, 0.7);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">value</span></div>
    </div><button data-v-55bfbadb="" type="button" class="v-btn v-theme--dark text-accent v-btn--density-default v-btn--size-small v-btn--variant-text text-body-1 pa-0 vd-data-list-item-action-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Action</span>
      <!---->
      <!---->
    </button>
  </div>
</li>
`;

exports[`DataListItem > renders correctly in row mode 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="vd-row">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">value</span></div>
    </div>
    <v-btn-stub data-v-55bfbadb="" color="accent" class="text-body-1 pa-0 vd-data-list-item-action-btn" symbol="Symbol(vuetify:v-btn-toggle)" flat="false" icon="false" block="false" slim="false" stacked="false" ripple="true" border="false" density="default" disabled="false" loading="false" tile="false" replace="false" exact="false" size="small" tag="button" variant="text"></v-btn-stub>
  </div>
</li>
`;

exports[`DataListItem > renders correctly value in a chip 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value">
        <v-chip-stub data-v-55bfbadb="" class="mt-1" closable="false" closeicon="$delete" closelabel="$vuetify.close" draggable="false" filter="false" filtericon="$complete" label="false" pill="false" ripple="true" modelvalue="true" border="false" density="default" disabled="false" tile="false" replace="false" exact="false" size="default" tag="span" variant="tonal"></v-chip-stub>
      </div>
    </div>
    <!--v-if-->
  </div>
</li>
`;

exports[`DataListItem > renders correctly with a NaN value 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">…</span></div>
    </div>
    <!--v-if-->
  </div>
</li>
`;

exports[`DataListItem > renders correctly with a number value 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">123</span></div>
    </div>
    <!--v-if-->
  </div>
</li>
`;

exports[`DataListItem > renders correctly with a value 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">value</span></div>
    </div>
    <!--v-if-->
  </div>
</li>
`;

exports[`DataListItem > renders correctly with an action 1`] = `
<li data-v-55bfbadb="" class="vd-data-list-item d-flex flex-wrap">
  <!--v-if-->
  <div data-v-55bfbadb="" class="vd-data-list-item-content">
    <div data-v-55bfbadb="" class="">
      <div data-v-55bfbadb="" class="vd-data-list-item-label text-caption" style="color: rgba(0, 0, 0, 0.6);">Test</div>
      <div data-v-55bfbadb="" class="vd-data-list-item-value"><span data-v-55bfbadb="" class="text-body-1">…</span></div>
    </div>
    <v-btn-stub data-v-55bfbadb="" color="accent" class="text-body-1 pa-0 vd-data-list-item-action-btn" symbol="Symbol(vuetify:v-btn-toggle)" flat="false" icon="false" block="false" slim="false" stacked="false" ripple="true" border="false" density="default" disabled="false" loading="false" tile="false" replace="false" exact="false" size="small" tag="button" variant="text"></v-btn-stub>
  </div>
</li>
`;
